cliconfig/config.go

```diff
@@ -29,20 +31,15 @@ var (
 	configDir = os.Getenv("DOCKER_CONFIG")
 )
 
-func getDefaultConfigDir(confFile string) string {
-	confDir := filepath.Join(homedir.Get(), confFile)
-	// if the directory doesn't exist, maybe we called docker with sudo
-	if _, err := os.Stat(configDir); err != nil {
-		if os.IsNotExist(err) {
-			return filepath.Join(homedir.GetWithSudoUser(), confFile)
-		}
-	}
-	return confDir
-}
-
 func init() {
 	if configDir == "" {
-		configDir = getDefaultConfigDir(".docker")
+		configDir = filepath.Join(homedir.Get(), defaultConfigDir)
+		if _, err := os.Stat(configDir); err != nil {
+			// if the directory doesn't exist, maybe we called docker with sudo
```

Comment:
Sorry, to reply to the first question you asked - the default is always the home dir of the user (even calling with sudo resolves to the home dir to first firstly, then fallback to the sudo user who called sudo if the dir isn't found)